qualification/Jenkinsfile

```
@@ -73,20 +73,33 @@ pipeline {
          script {
            sh 'pip install -r qualification/requirements.txt'
            sh 'pip install --no-deps ".[qualification]" && pip check'
-           try {
-             sh 'spead2_net_raw pytest -v -c qualification/pytest-jenkins.ini qualification --image-override katgpucbf:harbor.sdp.kat.ac.za/cbf/katgpucbf:latest'
-           }
-           finally {
-             COMMIT_ID = sh(script: 'qualification/report/generate_pdf.py report.json report.pdf -c', returnStdout: true)
-             sh(script: 'qualification/report/generate_pdf.py report.json procedure.pdf --generate-procedure-doc')
-             publishHTML(target: [keepAll: true, reportName: 'Qualification Test Report', reportDir: '', reportFiles: 'report.pdf, report.json', reportTitles: 'report.pdf, report.json'])
-             publishHTML(target: [keepAll: true, reportName: 'Qualification Test Procedure', reportDir: '', reportFiles: 'procedure.pdf', reportTitles: 'procedure.pdf'])
-             currentBuild.displayName = "#${BUILD_NUMBER} (${COMMIT_ID})"
-           }
+           sh 'spead2_net_raw pytest -v -c qualification/pytest-jenkins.ini qualification --suppress-tests-failed-exit-code --image-override katgpucbf:harbor.sdp.kat.ac.za/cbf/katgpucbf:latest --junitxml=result.xml'
          }
        }
-    }
-  }
+     }
+
+     stage('Publish results') {
+       steps {
+         script {
+           // Compress and publish json file so that at least we have that.
+           sh 'xz --keep --force -T 0 report.json'
+           publishHTML(target: [keepAll: true, reportName: 'Qualification Test Intermediate JSON', reportDir: '', reportFiles: 'report.json.xz', reportTitles: 'report.json.xz'])
+
+           // Generate and publish test report
+           COMMIT_ID = sh(script: 'qualification/report/generate_pdf.py report.json report.pdf -c', returnStdout: true)
+           currentBuild.displayName = "#${BUILD_NUMBER} (${COMMIT_ID})"
+           publishHTML(target: [keepAll: true, reportName: 'Qualification Test Report', reportDir: '', reportFiles: 'report.pdf', reportTitles: 'report.json'])
```

The reportTitles is wrong (and unnecessary)
```suggestion
           publishHTML(target: [keepAll: true, reportName: 'Qualification Test Report', reportDir: '', reportFiles: 'report.pdf'])
```